src/test/java/org/prebid/server/bidder/medianet/MedianetBidderTest.java

```diff
@@ -38,59 +45,137 @@ public void setup() {
+        assertThatIllegalArgumentException().isThrownBy(() -> new MedianetBidder("invalid_url", jacksonMapper));
     }
 
     @Test
-    public void makeHttpRequestsShouldNotModifyIncomingRequest() {
+    public void makeBidsShouldReturnBannerBidIfBannerIsPresentInRequestImp() throws JsonProcessingException {
         // given
-        final BidRequest bidRequest = givenBidRequest();
+        final HttpCall<BidRequest> httpCall = sampleHttpCall(givenBidRequest(impBuilder -> impBuilder
+                        .id("123")
+                        .banner(Banner.builder().build())),
+                mapper.writeValueAsString(
+                        givenBidResponse(bidBuilder -> bidBuilder.impid("123"))));
 
         // when
-        final Result<List<HttpRequest<BidRequest>>> result;
-        result = medianetBidder.makeHttpRequests(bidRequest);
+        final Result<List<BidderBid>> result = medianetBidder.makeBids(httpCall, null);
 
         // then
         assertThat(result.getErrors()).isEmpty();
         assertThat(result.getValue()).hasSize(1)
-            .extracting(httpRequest -> mapper.readValue(httpRequest.getBody(), BidRequest.class))
-            .containsExactly(bidRequest);
+                .contains(BidderBid.of(Bid.builder().impid("123").build(), banner, "USD"));
     }
 
     @Test
-    public void makeBidsShouldReturnErrorIfResponseBodyCouldNotBeParsed() {
+    public void makeBidsShouldReturnVideoBidIfVideoIsPresentInRequestImp() throws JsonProcessingException {
         // given
-        final HttpCall<BidRequest> httpCall = sampleHttpCall(givenBidRequest(), "invalid response");
+        final HttpCall<BidRequest> httpCall = sampleHttpCall(givenBidRequest(impBuilder -> impBuilder
+                        .id("123")
+                        .video(Video.builder().build())),
+                mapper.writeValueAsString(givenBidResponse(bidBuilder -> bidBuilder.impid("123"))));
+
+        // when
+        final Result<List<BidderBid>> result = medianetBidder.makeBids(httpCall, null);
+
+        // then
+        assertThat(result.getErrors()).isEmpty();
+        assertThat(result.getValue()).hasSize(1)
+                .contains(BidderBid.of(Bid.builder().impid("123").build(), video, "USD"));
+    }
+
+    @Test
+    public void makeBidsShouldReturnxNativeBidIfxNativeIsPresentInRequestImp() throws JsonProcessingException {
+        // given
+        final HttpCall<BidRequest> httpCall = sampleHttpCall(givenBidRequest(impBuilder -> impBuilder
+                        .id("123")
+                        .xNative(Native.builder().build())),
+                mapper.writeValueAsString(givenBidResponse(bidBuilder -> bidBuilder.impid("123"))));
+
+        // when
+        final Result<List<BidderBid>> result = medianetBidder.makeBids(httpCall, null);
+
+        // then
+        assertThat(result.getErrors()).isEmpty();
+        assertThat(result.getValue()).hasSize(1)
```

Comment:
no need for size check

Reply:
i think we can leave `hasSize`, it's more accuracy test or am i wrong ?

Reply:
when you pass single item into `containsExactly`, it checks if collection contains exactly one item and this item equals to passed one.

Reply:
understand, thanks